### PR TITLE
Add crystal shards support

### PR DESCRIPTION
**Pull request by Stygmates**

### CODE REVIEW

File: build/shards.zsh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+
+function shards.build {
+	write "${target}: ${src[@]} ${depends[$target]}"
+	write "\t@echo '${fg_bold[magenta]}  [SHARD]    ${fg_bold[white]}$target${reset_color}'"
+	write "\t${Q}shards build"
```
Comment (Lukc) [inline build/shards.zsh:5]: Is not related to the target. You could have a `make foo` that doesn’t build foo.

File: build/shards.zsh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+function shards.uninstall {
+	binary.uninstall "$@"
+}
+
+function shards.clean {
+	script.clean "$@"
+}
+
+function shards.distfiles {
+	echo "${target}"
```
Comment (Lukc) [inline build/shards.zsh:23]: shards will rely on things from lib/ and its shard.yml. lib/ can be ignored if you run `shards update` somewhere, but shard.yml will be needed, and only once.

**Review by Lukc — CHANGES_REQUESTED** [review]

**Comment by Lukc** [comment]
Hi @Stygmates! Are you still working on this? Can I close the PR, or are you still interested in having native shards support?

**Comment by Stygmates** [comment]
I'm not working on this anymore, closing this PR.

**Pull request closed without merging**